src/o365/spo/commands/theme/theme-apply.ts

```diff
@@ -37,9 +50,14 @@ class SpoThemeApplyCommand extends SpoCommand {
   }
 
   public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const isSharePointTheme: boolean = args.options.sharePointTheme ? true : false;
+    const resource: string = isSharePointTheme ? Auth.getResourceFromUrl(args.options.webUrl) : auth.service.resource;
+    let siteAccessToken: string = '';
+
     auth
-      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .getAccessToken(resource, auth.service.refreshToken as string, cmd, this.debug)
       .then((accessToken: string): Promise<ContextInfo> => {
+        siteAccessToken = accessToken;
```

Comment:
Since we don't need request digest for applying a SharePoint theme, we could optimize this part by resolving the promise without retrieving request digest.